src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,183 @@
+package duke;
+
+import exceptions.EmptyNameException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Stores the task in a file under /data/duke.txt
+ * Contains the methods to load the file and modify the file when the tasks are modified in the app
+ */
+public class Storage {
+    private final String path;
+    private final TaskList taskList;
+
+    /**
+     * Constructor for the Storage class
+     *
+     * @param path     path of the file
+     * @param taskList current Tasklist state
+     */
+    public Storage(String path, TaskList taskList) {
+        this.path = path;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Adds text to the file
+     *
+     * @param filePath  path of the file
+     * @param textToAdd text to be added to the file
+     * @throws IOException If file is not found
+     */
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true);
+        fw.write(textToAdd + "\n");
+        fw.close();
+    }
+
+    /**
+     * Loads the file into the app so that a saved state can be retrieved
+     * Creates file and folder if it doesn't exist
+     *
+     * @throws IOException Throws exception when file is missing
+     */
+    public void loadFile() throws IOException {
+        File file = new File(this.path);
+        File parentFile = file.getParentFile();
+
+        if (parentFile != null && !parentFile.exists() && !parentFile.mkdirs()) {
+            throw new IllegalStateException("Error creating directory" + parentFile);
+        }
+        file.createNewFile();
+
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            taskList.readTaskFromLoader(line);
+        }
+
+        scanner.close();
+
+    }
+
+    /**
+     * Takes in command from User and handle the logic on what to save onto the file
+     *
+     * @param command Command received from User
+     */
+    public void addLineToFile(String command) {
+        try {
+            String[] combiStr;
+            String[] split = command.split(" ", 2);
```

Comment:
I have noticed the variable name `split` and the method `split` has the same name. To prevent confusion I suggest changing the variable name to `splitCommand`.